app/controllers/spree/admin/insights_controller.rb

```diff
@@ -103,8 +103,8 @@ def set_default_pagination
           @pagination_hash = { paginate: false }
           unless params[:paginate] == 'false'
             @pagination_hash[:paginate] = true
-            @pagination_hash[:records_per_page] = params[:per_page].try(:to_i) || Spree::Config[:records_per_page]
-            @pagination_hash[:offset] = params[:page].to_i * @pagination_hash[:records_per_page]
+            @pagination_hash[:records_per_page] = params[:per_page].try(:to_i) || 25
```

Comment:
why removing `Spree::Config[:records_per_page]` ? 
you even create it in `app/models/spree/app_configuration_decorator.rb`

Reply:
Good catch, I removed it before the AppConfigurationDecorator was refactored. Before that, it didn't work.